sort/patiencesort.go

```diff
@@ -0,0 +1,90 @@
+// Package sort
+// Patience sorting is a sorting algorithm inspired by the card game patience.
+//
+// For more details check out those links below here:
+// GeeksForGeeks article : https://www.geeksforgeeks.org/patience-sorting/
+// Wikipedia article: https://en.wikipedia.org/wiki/Patience_sorting
+// authors [guuzaa](https://github.com/guuzaa)
+// see patiencesort.go
+package sort
+
+import "github.com/TheAlgorithms/Go/constraints"
+
+func Patience[T constraints.Ordered](arr []T) []T {
+	if len(arr) <= 1 {
+		return arr
+	}
+
+	var piles [][]T
+
+	// Traverse the given array
+	for _, card := range arr {
+		// Iterate over all the piles
+		// and check the top of the stack of each pile
+		// is less than the current element or not.
+		left, right := 0, len(piles)
+		for left < right {
+			mid := left + (right-left)>>1
```

Comment:
```suggestion
			mid := (left + (right-left))>>1
```

might make it more clear what you're trying to do.

But I also suggest to keep this simple, no need for fancy bit shifting when we already know that the compiler is quick enough in division by 2. So
```suggestion
			mid := (left + (right-left))/2
```
might be more suitable.